components/browser/state/src/main/java/mozilla/components/browser/state/state/SessionState.kt

```diff
@@ -24,4 +24,17 @@ interface SessionState {
     val engineState: EngineState
     val extensionState: Map<String, WebExtensionState>
     val contextId: String?
+
+    /**
+     * Copy the class and override some parameters.
+     */
+    @Suppress("LongParameterList")
+    fun copySession(
```

Comment:
I wish we could make this `internal` since we only need this inside the reducers. But I guess this only possible with an extension method that does instance checks and this is awkward as well.

nit: I feel like something like `createCopy()` would be a better name. Calling `copySession()` on `SessionState/TabSessionState/CustomTabSessionState` would make me wonder what Session I am copying where.